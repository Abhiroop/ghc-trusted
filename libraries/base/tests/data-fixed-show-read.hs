
module Main (main) where

import Data.Fixed

data B7

instance HasResolution B7 where
  resolution _ = 128

main :: IO ()
main = do doit 38.001
          doit 38.009
          doit 38.01
          doit 38.09
          print (read "38" :: Centi)
          doit (-38.001)
          doit (-38.009)
          doit (-38.01)
          doit (-38.09)
          print (read "-38" :: Centi)
          print (read "0.008" :: Fixed B7)
          print (read "-0.008" :: Fixed B7)
          print $ show (Just (-1 :: Milli))
          print $ show (Just (1 :: Milli))
          print ((read $ show (Just (-1 :: Deci))) :: Maybe Deci)
          print ((read $ show (Just (1 :: Deci))) :: Maybe Deci)

doit :: Centi -> IO ()
doit c = do let s = show c
                r = read s :: Centi
            putStrLn s
            print r
