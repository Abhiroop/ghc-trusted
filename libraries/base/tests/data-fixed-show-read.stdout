38.00
38.00
38.00
38.00
38.01
38.01
38.09
38.09
38.00
-38.00
-38.00
-38.00
-38.00
-38.01
-38.01
-38.09
-38.09
-38.00
0.008
-0.008
"Just (-1.000)"
"Just 1.000"
Just (-1.0)
Just 1.0
